### PR TITLE
Add compiler detection from cabal file.

### PR DESCRIPTION
**Pull request by rpglover64**
Code should probably be refactored to be less redundant, given that there are now 3 uses of very similar constructs.

### CODE REVIEW

**Comment by lunaryorn** [comment]
@rpglover64 Thanks for this PR.  I have a couple of questions and remarks, which I've added as comments to the diff.  Would you mind to address them?

File: flycheck-haskell.el [hunk 1]
```diff
@@ -143,6 +143,9 @@ Return the configuration."
   (or (flycheck-haskell-get-cached-configuration cabal-file)
       (flycheck-haskell-read-and-cache-configuration cabal-file)))
 
+(defconst flycheck-haskell-cabal-config "cabal.config"
```
Comment (lunaryorn) [inline flycheck-haskell.el]: Shouldn't we look at `cabal.sandbox.config` as well?  Currently it doesn't contain this setting, but it probably might do so in future.

File: flycheck-haskell.el [hunk 2]
```diff
@@ -164,6 +167,33 @@ database was not found."
...
+  "Regular expression to parse the compiler path.")
+
+(defun flycheck-haskell-get-compiler (cabal-config)
+  "Get the compiler path from CABAL-CONFIG.
+
+Return the compiler path as string, or nil, if the database was
+not found."
+  (with-temp-buffer
+    (insert-file-contents cabal-config)
+    (goto-char (point-min))
```
Comment (lunaryorn) [inline flycheck-haskell.el]: We do already require `haskell-cabal`, so you can just use its `haskell-cabal-get-setting` here, e.g. `(haskell-cabal-get-setting "with-compiler")`

File: flycheck-haskell.el (hunk 1 above)
Comment (rpglover64) [inline flycheck-haskell.el]: I'd cross that bridge when it comes up, especially since I don't think it's coming as soon as we'd like.

Comment (lunaryorn) [inline flycheck-haskell.el]: @rpglover64 Ok, you're the expert, you make the choices.  As a matter of fact, I didn't even know that `cabal.config` existed until you made this PR ;)

Comment (rpglover64) [inline flycheck-haskell.el]: > Ok, you're the expert

Not nearly, but thanks for the vote of confidence! TBH, I didn't know `cabal.config` existed either, until I needed the `with-compiler` setting.

Comment (lunaryorn) [inline flycheck-haskell.el]: @rpglover64 Hah, still, you're ahead of me :)

File: flycheck-haskell.el (hunk 2 above)
Comment (rpglover64) [inline flycheck-haskell.el]: I'm not sure that's correct... Reading through the `haskell-cabal` source, it looks like `haskell-cabal-get-setting` is meant to handle `*.cabal` files, but project-specific `cabal.config` files are in the format of `~/.cabal/config`.
Also, I can't seem to get it to work (although it does when I call it interactively).

**Comment by rpglover64** [comment]
Ideally, we'd also add a test for this functionality, but I couldn't really follow how the tests are written (I'm not that proficient in Emacs development).

**Comment by lunaryorn** [comment]
@rpglover64 Merged and released as 0.7.  Thanks for this feature!

**Pull request merged**